src/sidebar/annotation-metadata.js

```diff
@@ -101,6 +101,24 @@ function isNew(annotation) {
   return !annotation.id;
 }
 
+/** Return `true` if the given annotation is public, `false` otherwise. */
+function isPublic(annotation) {
```

Comment:
This returns false if _any_ of the principals that can read an annotation are an individual (`acct:...`) as opposed to a group (`group:...`).

However, the logic we want is that the annotation is public if _at least one of_ the principals that can read an annotation is a group _which is readable by everyone_. This means that eventually we will need to consider both the group's permissions as well as the annotation's principals to avoid including annotations in private groups. Initially however the only group shown on pages using publisher accounts will be the world-readable publisher group, so we can just check that _at least one_ of the annotation's principals is a group.